### DISCUSSION TITLE
[前端代码] 建议去掉前端打包时生成的gzip文件

### DISCUSSION THREAD

**Discussion by bddjr** [1Panel-dev/1Panel#5599] [General]
目前go语言已有的gzip框架都不支持使用静态gzip文件，打包这个纯属浪费硬盘寿命，建议移除以下内容

https://github.com/1Panel-dev/1Panel/blob/dev/frontend/vite.config.ts#L59C13-L66C20

**Comment by zhengkunwang223** [comment]
感谢建议 我们会在后续打包的时候去掉 gzip 压缩
等后续解决了前端文件的压缩访问 再打开

**Reply by bddjr** [reply]
后端不太可能适配静态gz文件的，因为如果要实现这个功能，意味着不能使用go标准库，需要自己重写StaticFile方法，没人愿意干这活。

**Reply by zhengkunwang223** [reply]
> 后端不太可能适配静态gz文件的，因为如果要实现这个功能，意味着不能使用go标准库，需要自己重写StaticFile方法，没人愿意干这活。

初步的打算是 前端请求.js 文件 后端如果匹配到 .js.gzip 文件，就返回，并且修改  Header 让浏览器解压

**Reply by bddjr** [reply]
你们有这个闲工夫写这个，也是神人了

**Reply by bddjr** [reply]
没有什么证明动态gzip会占用过多的cpu，所以暂时没有必要。一些网站的静态gzip是直接借助nginx实现的

**Reply by bddjr** [reply]
相关代码在http.StaticFile里是写死的，gin封装的也只是调用了前者，  

在调用之前改Header没用，因为StaticFile会覆盖。  
调用之后改Header没用，因为Header已经被锁定了，改不了。  

所以要实现这个只能重写StaticFile，但是这太复杂了，没人愿意干。

**Comment by bddjr** [comment]
Merged #5606

**Comment by bddjr** [comment]
我写了个模组，实现了读取静态gzip和br文件的功能

https://github.com/bddjr/gzipstatic-gin

**Reply by bddjr** [reply]
但是我懒得提交PR了。之前我F12抓包的时候发现他们私自把Last-Modified响应头移除了，这样的产品是不会真正在意带宽占用的。
